opensrp-core/src/main/java/org/smartregister/view/interactor/StatsFragmentInteractor.java

```diff
@@ -108,40 +128,71 @@ public void run() {
-            syncInfoMap.put(TASK_UNPROCESSED_EVENTS, cursor.getInt(0));
+            syncInfoMap.put(TASK_UNPROCESSED_EVENTS, String.valueOf(cursor.getInt(0)));
         } else if (syncStatus == null) {
-            syncInfoMap.put(NULL_EVENT_SYNC_STATUS, cursor.getInt(0));
+            syncInfoMap.put(NULL_EVENT_SYNC_STATUS, String.valueOf(cursor.getInt(0)));
         }
     }
 
     private void populateClientSyncInfo(Cursor cursor) {
         String syncStatus = cursor.getString(cursor.getColumnIndex(SYNC_STATUS));
         if (BaseRepository.TYPE_Synced.equals(syncStatus)) {
-            syncInfoMap.put(SYNCED_CLIENTS, cursor.getInt(0));
+            syncInfoMap.put(SYNCED_CLIENTS, String.valueOf(cursor.getInt(0)));
         } else if (BaseRepository.TYPE_Unsynced.equals(syncStatus)) {
-            syncInfoMap.put(UNSYNCED_CLIENTS, cursor.getInt(0));
+            syncInfoMap.put(UNSYNCED_CLIENTS, String.valueOf(cursor.getInt(0)));
         }
     }
 
     private void populateValidatedEventsInfo(Cursor cursor) {
         String syncStatus = cursor.getString(cursor.getColumnIndex(VALIDATION_STATUS));
         if (BaseRepository.TYPE_Valid.equals(syncStatus)) {
-            syncInfoMap.put(VALID_EVENTS, cursor.getInt(0));
+            syncInfoMap.put(VALID_EVENTS, String.valueOf(cursor.getInt(0)));
         } else if (BaseRepository.TYPE_InValid.equals(syncStatus)) {
-            syncInfoMap.put(INVALID_EVENTS, cursor.getInt(0));
+            syncInfoMap.put(INVALID_EVENTS, String.valueOf(cursor.getInt(0)));
         }
     }
 
     private void populateValidatedClientsInfo(Cursor cursor) {
         String validationStatus = cursor.getString(cursor.getColumnIndex(VALIDATION_STATUS));
         if (BaseRepository.TYPE_Valid.equals(validationStatus)) {
-            syncInfoMap.put(VALID_CLIENTS, cursor.getInt(0));
+            syncInfoMap.put(VALID_CLIENTS, String.valueOf(cursor.getInt(0)));
         } else if (BaseRepository.TYPE_InValid.equals(validationStatus)) {
-            syncInfoMap.put(INVALID_CLIENTS, cursor.getInt(0));
+            syncInfoMap.put(INVALID_CLIENTS, String.valueOf(cursor.getInt(0)));
         }
     }
+
+    public void populateUserInfo() {
+        AllSharedPreferences sharedPreferences = CoreLibrary.getInstance().context().userService().getAllSharedPreferences();
+        syncInfoMap.put(USER_NAME, sharedPreferences.fetchRegisteredANM());
+        syncInfoMap.put(USER_TEAM, sharedPreferences.fetchDefaultTeam(sharedPreferences.fetchRegisteredANM()));
+        syncInfoMap.put(USER_LOCALITY, sharedPreferences.fetchCurrentLocality());
+    }
+
+
+    private void populateBuildInfo() {
+        try {
+            syncInfoMap.put(APP_VERSION_NAME, getBuildConfigValue("VERSION_NAME"));
+            syncInfoMap.put(APP_VERSION_CODE, getBuildConfigValue("VERSION_CODE"));
+            syncInfoMap.put(DB_VERSION, getBuildConfigValue("DATABASE_VERSION"));
```

Comment:
See earlier comment on `DB_VERSION`

Reply:
Refactored implementation to return a default value